content/articles/spring-webflux/index.md

```diff
@@ -0,0 +1,315 @@
+### Application Setup
+We are going to use [spring initialr](https://start.spring.io/) to generate our application startup code.
+1. On your web browser, navigate to [spring initialr](https://start.spring.io/).
+2. Input the group as `io.section and name as `webfluxexample`.
+3. Add `Spring webflux`,`Mongo reactive`, and `lombok` as project dependencies.
+4. CLick generate to download the startup project files as a zip.
+5. Extract the zip file and open the project in your favorite code editor or IDE.
+6. Add dependencies below to `pom.xml` file.
+   ```xml
+   <dependencies>
+     <dependency>
+        <groupId>javax.xml.bind</groupId>
+        <artifactId>jaxb-api</artifactId>
+        <version>2.3.0</version>
+     </dependency>
+     <dependency>
+        <groupId>javax.servlet</groupId>
+        <artifactId>javax.servlet-api</artifactId>
+        <version>3.1.0</version>
+        <scope>provided</scope>
+     </dependency>
+   </dependencies>
+   ```
+
+### Configuration layer
+1. Create a package named `config` in the project root package.
+2. Create a file named `ApplicationConfig.java` and add the code snippet below.
+    
+```java
+@Configuration //Marks the class as a configuration class
+public class ApplicationConfig {
+    //Method below sets the path to the application.properties file.
+    @Bean
+    public PropertyPlaceholderConfigurer placeholderConfigurer() {
+        PropertyPlaceholderConfigurer configurer = new PropertyPlaceholderConfigurer();
+        configurer.setLocation(new ClassPathResource("application.properties"));
+        configurer.setIgnoreUnresolvablePlaceholders(true);
+        return configurer;
+    }
+}
+```
+ - The code snippet above sets the path to the `application.properties` file, it contains the project configurations.
+
+3. Create a new file named `MongoConfig.java` in the `config` package we created earlier and the code snippet below.
+   
+```java
+@Configuration
+@EnableMongoRepositories(basePackages = "io.section.webfluxexample.repositories")
+public class MongoDBConfig extends AbstractReactiveMongoConfiguration {
+
+    @Value("${database.name}")//Gets the database name from application.properties
+    private String databaseName;
+    //Mongodb connection string. Replace <username> and <password> with your mongodb username and password
+    private String name = "mongodb+srv://<username>:<password>@cluster0.mk0n7.gcp.mongodb.net/myFirstDatabase?retryWrites=true&w=majority";
+
+    @Override
+    protected String getDatabaseName() {
+        return databaseName;
+    }
+
+    @Override
+    public MongoClient reactiveMongoClient() {
+        return MongoClients.create(name);
+    }
+
+    @Bean
+    public ReactiveMongoTemplate reactiveMongoTemplate() {
+        return new ReactiveMongoTemplate(reactiveMongoClient(), getDatabaseName());
+    }
+}
+
+```
+> For instructions on creating MongoDB collection in Mongo atlas, read [Spring Data and Mongodb](/engineering-education/spring-data-mongodb/).
```

Comment:
Link to the actual web page that article is on, assuming it is published.

Reply:
In one of the open PRs @hectorkambow said when referring to articles published on section.io, we don't need to include the full URL to the article.

Reply:
Oh okay, I actually never knew that thanks for letting me know. I think I might have done it before with the full URL and got away with it that's strange.